src/net_processing.cpp

```diff
@@ -3247,6 +3290,55 @@ bool PeerLogicValidation::SendMessages(CNode* pto, std::atomic<bool>& interruptM
             }
         }
 
+        // Check that outbound peers have reasonable chains
+        if (!(state.fProtectFromDisconnect || pto->fInbound || pto->m_manual_connection || pto->fFeeler || pto->fOneShot) && state.fSyncStarted) {
```

Comment:
The check `pto->fInbound || pto->m_manual_connection || pto->fFeeler || pto->fOneShot` is done twice in this PR. Perhaps this check can be put into a separate function like:
```
bool canBeProtectedFromDisconnect(CNode* node)
```
This would also make the code a bit more readable.